### ISSUE TITLE
compatibility with SonarQube 5.1

### ISSUE THREAD

**Issue by zenderol** [arthepsy/sonar-scala-extra#2]
After installing the sonar-scapegoat-plugin-1.1 on SonarQube 5.1 ,I get the following exception during the sonar runner analysis regardles if I try to upload the scapegoat report or not. 

```

>> ERROR: Error during Sonar runner execution
>> 
>> org.sonar.runner.impl.RunnerException: Unable to execute Sonar
>> 
>> at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:91)
>> 
>> at org.sonar.runner.impl.BatchLauncher$1.run(BatchLauncher.java:75)
>> 
>> at java.security.AccessController.doPrivileged(Native Method)
>>      at org.sonar.runner.impl.BatchLauncher.doExecute(BatchLauncher.java:69)
>>      at org.sonar.runner.impl.BatchLauncher.execute(BatchLauncher.java:50)
>>      at org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:102)
>>      at org.sonar.runner.api.Runner.execute(Runner.java:100)
>>      at org.sonar.runner.Main.executeTask(Main.java:70)
>> 
>> at org.sonar.runner.Main.execute(Main.java:59)
>> 
>> at org.sonar.runner.Main.main(Main.java:53)
>> 
>> Caused by: java.lang.NoSuchMethodError: org.sonar.api.batch.sensor.SensorDescriptor.dependsOn([Lorg/sonar/api/batch/measure/Metric;)Lorg/sonar/api/batch/sensor/SensorDescriptor;
>> 
>> at eu.arthepsy.sonar.plugins.scapegoat.rule.ScapegoatReportSensor.describe(ScapegoatReportSensor.java:80)
>> 
>> at org.sonar.batch.scan.SensorWrapper.<init>(SensorWrapper.java:44)
>> 
>> at org.sonar.batch.bootstrap.BatchExtensionDictionnary.getFilteredExtensions(BatchExtensionDictionnary.java:102)
>> at org.sonar.batch.bootstrap.BatchExtensionDictionnary.select(BatchExtensionDictionnary.java:68)
>>      at org.sonar.batch.phases.SensorsExecutor.execute(SensorsExecutor.java:47)
>>      at org.sonar.batch.phases.DatabaseModePhaseExecutor.execute(DatabaseModePhaseExecutor.java:120)
>>      at org.sonar.batch.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:264)
>> at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:92)
>> at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:77)
>>      at org.sonar.batch.scan.ProjectScanContainer.scan(ProjectScanContainer.java:235)
>>      at org.sonar.batch.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:230)
>>      at org.sonar.batch.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:220)
>>      at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:92)
>>      at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:77)
>>      at org.sonar.batch.scan.ScanTask.scan(ScanTask.java:57)
>>      at org.sonar.batch.scan.ScanTask.execute(ScanTask.java:45)
>> 
>> at org.sonar.batch.bootstrap.TaskContainer.doAfterStart(TaskContainer.java:135)
>> 
>> at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:92)
>> 
>> at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:77)
>> 
>> at org.sonar.batch.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:158)
>> 
>> at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:95)
>> 
>> at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:67)
>> 
>> at org.sonar.runner.batch.IsolatedLauncher.execute(IsolatedLauncher.java:48)
>> 
>> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>> at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>>      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>>      at java.lang.reflect.Method.invoke(Method.java:497)
>> at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:87)
>>      ... 9 more
>> Return code: 1.
```

I've installed the required [Sonar-Scala](https://github.com/arthepsy/sonar-scala)  plugin, so my only idea is that SonarQube 5.1+ is not supported.

**Comment by arthepsy** [comment]
Thanks for report, will look into it. I have tried it on 5.0.x, but haven't jet tried on 5.1+.  Which exactly version of 5.1.x branch are using?

**Comment by zenderol** [comment]
I'm evaluating your plugin with the official docker image of [sonarqube](https://github.com/SonarSource/docker-sonarqube) (5.1 latest) . 

This is the list of all sonar plugins I installed so far:

```
Java [java]     3.5     
JavaScript [javascript]     2.8     
Scala [scala]   1.0.1-SNAPSHOT  
Scapegoat [scapegoat]   1.1
Git [scmgit]    1.0     
SVN [scmsvn]    1.0     
Scoverage [scoverage]   5.1.1
```

 As soon as  I install  your sonar-scapegoat-plugin I receive the mentioned exception during a sonar-runner run, regardles if I'm using your plugin or not, which is strange.
I really hope we can find a fix for this problem!

**Comment by arthepsy** [comment]
I tested and can verify that it's broken with 5.1+, because of plugin API changes. I am trying to fix so it could support both 4.x and 5.x, but so far it seems that I will have to create two separate versions. In any case, this is fixable, just looking for best solution and will update once it's done.

**Comment by zenderol** [comment]
thank you for your effort!

**Comment by arthepsy** [comment]
After some thought, I created 4.5 - 5.1 compatible version. Issue should be fixed. @zenderol, could You please try with latest snapshot and give feedback? If everything will be fine, I will release new version.

**Comment by zenderol** [comment]
I will do that tomorrow!

**Comment by zenderol** [comment]
I can confirm that 08eda3768c0fdce0c5d0e2358aa66836fe7dba88 works as expected with SonarQube 5.1.latest
big thanks!

**Comment by arthepsy** [comment]
Thanks for confirmation. Released as v1.2.
